pkg/bloomcompactor/bloomcompactor.go

```diff
@@ -106,141 +145,273 @@ func (c *Compactor) stopping(_ error) error {
-	// create bloom filters from that.
-	bloom.Add([]byte(fmt.Sprint(data)))
+// TODO Revisit this step once v1/bloom lib updated to combine blooms in the same series
+func buildBloomBlock(bloomForChks v1.SeriesWithBloom, series Series) (bloomshipper.Block, error) {
+	localDst := createLocalFileName(series)
 
-	// block and seriesList
-	seriesList := []v1.SeriesWithBloom{
-		{
-			Series: series,
-			Bloom:  &bloom,
-		},
+	//write bloom to a local dir
+	builder, err := v1.NewBlockBuilder(v1.NewBlockOptions(), v1.NewDirectoryBlockWriter(localDst))
+	if err != nil {
+		level.Info(util_log.Logger).Log("creating builder", err)
+		return bloomshipper.Block{}, err
 	}
 
-	writer := v1.NewDirectoryBlockWriter(dst)
+	err = builder.BuildFrom(v1.NewSliceIter([]v1.SeriesWithBloom{bloomForChks}))
+	if err != nil {
+		level.Info(util_log.Logger).Log("writing bloom", err)
+		return bloomshipper.Block{}, err
+	}
 
-	builder, err := v1.NewBlockBuilder(
-		v1.BlockOptions{
-			SeriesPageSize: 100,
-			BloomPageSize:  10 << 10,
-		}, writer)
+	blockFile, err := os.Open(filepath.Join(localDst, "bloom"))
 	if err != nil {
-		return err
+		level.Info(util_log.Logger).Log("reading bloomBlock", err)
 	}
-	// BuildFrom closes itself
-	err = builder.BuildFrom(v1.NewSliceIter[v1.SeriesWithBloom](seriesList))
+
+	// read the checksum
+	blockFile.Seek(-4, 2)
+	checksum := make([]byte, 4)
+	blockFile.Read(checksum)
+
+	// Reset back to beginning
+	blockFile.Seek(0, 0)
+
+	objectStoreDst := createObjStorageFileName(series, fmt.Sprint(binary.BigEndian.Uint32(checksum)))
+
+	blocks := bloomshipper.Block{
+		BlockRef: bloomshipper.BlockRef{
+			Ref: bloomshipper.Ref{
+				TenantID:       series.tenant,
+				TableName:      series.tableName,
+				MinFingerprint: uint64(series.fingerPrint), //TODO will change once we compact multiple blooms into a block
+				MaxFingerprint: uint64(series.fingerPrint),
+				StartTimestamp: series.from.Unix(),
+				EndTimestamp:   series.through.Unix(),
+				Checksum:       binary.BigEndian.Uint32(checksum),
+			},
+			IndexPath: series.indexPath, // For reviewer, this is the same filepath with the original TSDBindex for this series?
+			BlockPath: objectStoreDst,
+		},
+		Data: blockFile,
+	}
+
+	return blocks, nil
+}
+
+func listSeries(ctx context.Context, objectClient client.ObjectClient) ([]Series, error) {
+	// Returns all the TSDB files, including subdirectories
+	prefix := "index/"
+	indices, _, err := objectClient.List(ctx, prefix, "")
+
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	// TODO Ask Owen, shall we expose a method to expose these paths on BlockWriter?
-	indexPath := filepath.Join(dst, "series")
-	bloomPath := filepath.Join(dst, "bloom")
+	var result []Series
+
+	for _, index := range indices {
+		s := strings.Split(index.Key, "/")
+
+		if len(s) > 3 {
+			tableName := s[1]
+
+			if !strings.HasPrefix(tableName, "loki_") || strings.Contains(tableName, "backup") {
```

Comment:
These should also also be not defined as constants here (they're both configurable). The file name prefix is set [here](https://github.com/grafana/loki/blob/main/pkg/storage/config/schema_config.go#L512). The `backup` is also not something we can reliably ignore here (it's feasible to name the table prefix "backup"). Instead, try using a `PeriodConfig` as an argument to this function.

It might also be worthwhile to build a file discovery method into the index shipper's API since it already does a lot of this logic internally.

Reply:
same reasoning as above